pkg/scheduler/scheduler.go

```diff
@@ -486,6 +496,19 @@ func (s *Scheduler) getInitialAssignments(log logr.Logger, wl *workload.Info, sn
 	return fullAssignment, nil
 }
 
+func (s *Scheduler) evictWorkloadAfterFailedTASReplacement(ctx context.Context, log logr.Logger, wl *kueue.Workload) error {
+	log.V(2).Info("Evicting workload after failed try to find a node replacement; TASFailedNodeReplacementFailFast enabled")
+	msg := fmt.Sprintf("Workload was evicted as there was no replacement for a failed node: %s", workload.NodeToReplace(wl))
+	if err := workload.EvictWorkload(ctx, s.client, s.recorder, wl, kueue.WorkloadEvictedDueToNodeFailures, msg, s.clock); err != nil {
+		return err
+	}
+	if err := workload.RemoveAnnotation(ctx, s.client, wl, kueuealpha.NodeToReplaceAnnotation); err != nil {
+		log.V(2).Error(err, fmt.Sprintf("Failed to remove annotation for node replacement %s", kueuealpha.NodeToReplaceAnnotation))
+		return err
```

Comment:
let's just return `fmt.Errorf("Failed to remove annotation for node replacement %s", kueuealpha.NodeToReplaceAnnotation)` as error to avoid double-logging

Reply:
Done